crates/astria-composer/src/bundler.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{
+    ds::RollupTxExt,
+    strategy::{NoStrategy, Strategy},
+};
+use color_eyre::eyre::{self, Context};
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+use tokio::sync::mpsc as tokio_mpsc;
+
+/// The Bundler module bundles received transactions based on the various
+/// MEV strategies that are implemented.
+/// It then forwards the bundles to the Executor
+pub(crate) struct Bundler {
+    collector_receiver: UnboundedReceiver<RollupTxExt>,
+    executor_sender: UnboundedSender<Vec<RollupTxExt>>,
+}
+
+impl Bundler {
+    pub(crate) fn new(
+        collector_receiver: UnboundedReceiver<RollupTxExt>,
+    ) -> (Self, UnboundedReceiver<Vec<RollupTxExt>>) {
+        let (stream, sink) = tokio_mpsc::unbounded_channel::<Vec<RollupTxExt>>();
+        let new_strategy = Self {
+            collector_receiver,
+            executor_sender: stream,
+        };
+        (new_strategy, sink)
+    }
+
+    async fn bundle_transactions(
+        rollup_txs: Vec<RollupTxExt>,
+    ) -> Result<Vec<RollupTxExt>, eyre::Error> {
+        // NOTE: Replace this with arbitrary strategy logic
+        let bundle = NoStrategy::execute(rollup_txs).await?;
```

Comment:
I dont think that this is the correct way to make this replaceable by others' strategies - they should instead replace the entire bundler by their own implementation for a strategy

Reply:
I'm not sure why? They should have to replace the bundler since that is just infra wiring - I would really rather make the strategies a separate trait that can be extended